### ISSUE TITLE
Support for Column-Specific Float Precision

### ISSUE THREAD

**Issue by vdreynol** [pola-rs/polars#14766]
### Description

**Proposal:**
`write_csv() & sink_csv()` allow you to specify a single precision to apply to all float columns.

Update this behavior to support dictionaries of `{'col': precision}` to allow for column-specific formatting.

**Background & Use Case:**
I'm currently using polars for ETL in a legacy environment, and the downstream tooling expects columns to have the same precision as the database. `NUMERIC(19,6)` expects 6 decimal places, `NUMERIC(5,3)` expects 3 & so on.

This is related to #11929 & #7133 , but a bit of a narrower request.

**Comment by JulianCologne** [comment]
I see you point and it makes sense!

Ideally you would use the `Decimal` type which maps exactly to those database types with a specific precision and scale!
Unfortunately this is still experimental and has lots of limitations and I just checked and its not possible to write a Decimal to csv 😆

**Comment by vdreynol** [comment]
Yes @JulianCologne , I agree that that would be the ideal solution.

**Comment by cmdlineluser** [comment]
@Julian-J-S  Just stumbled upon this issue - it seems your example now works (and `sink_csv` also).
